core/src/jsMain/kotlin/Peripheral.kt

```diff
@@ -205,11 +205,13 @@ public class JsPeripheral internal constructor(
     ): BluetoothRemoteGATTCharacteristic {
         val services = checkNotNull(platformServices) { "Services have not been discovered for $this" }
         val characteristics = services
-            .first { it.serviceUuid == characteristic.serviceUuid }
-            .characteristics
+            .firstOrNull { it.serviceUuid == characteristic.serviceUuid }
+            ?.characteristics
+            ?: throw IOException("Service ${characteristic.serviceUuid} not found")
         return characteristics
-            .first { it.characteristicUuid == characteristic.characteristicUuid }
-            .bluetoothRemoteGATTCharacteristic
+            .firstOrNull { it.characteristicUuid == characteristic.characteristicUuid }
+            ?.bluetoothRemoteGATTCharacteristic
+            ?: throw IOException("Characteristic ${characteristic.characteristicUuid} not found")
```

Comment:
This is unrelated - was added when debugging, can remove if not wanted

Reply:
Ya, please revert. We'll make this change (or similar) when we close #38.